src/main/java/ca/concordia/soen/ThrowsGenericVisitor.java

```diff
@@ -0,0 +1,72 @@
+package ca.concordia.soen;
+
+import org.eclipse.jdt.core.dom.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ThrowsGenericVisitor  extends ASTVisitor {
+    private final CompilationUnit compilationUnit;
+    int AntiPatternOccurrencesCount = 0;
+    List<AntiPatternOccurrence> ThrowsGenericOcurrencesList = new ArrayList<>();
+
+    public ThrowsGenericVisitor(CompilationUnit compilationUnit) {
+        this.compilationUnit = compilationUnit;
+    }
+
+    @Override
+    public boolean visit(MethodDeclaration node) {
+        List<Type> thrownExceptions = node.thrownExceptionTypes();
+        CompilationUnit cu = compilationUnit;
+        Block body = node.getBody();
+        if(thrownExceptions.size() == 1){
+            if (thrownExceptions.get(0) instanceof SimpleType && thrownExceptions.get(0).toString().equals("Exception")) {
+                antiPatternDetect(node, cu, thrownExceptions.get(0).getStartPosition());
```

Comment:
You can add a return true after this line, given that you don't need to run it until the end